### PR TITLE
build: add golang modules for apis

### PR DESCRIPTION
**Pull request by subhamkrai**
This change is beneficial to other projects that use kubernetes-csi-addons API and want to keep import dependency minimal. Importing API only reduces imported dependencies.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 
+replace github.com/csi-addons/kubernetes-csi-addons/apis => ./apis
+
 go 1.20
 
 require (
 	github.com/container-storage-interface/spec v1.8.0
-	github.com/csi-addons/spec v0.2.1-0.20230606140122-d20966d2e444
+	github.com/csi-addons/kubernetes-csi-addons/apis v0.0.0-00010101000000-000000000000
+	github.com/csi-addons/spec v0.2.0
```
Comment (nixpanic) [inline go.mod:10]: Please prevent this downgrade of the csi-addons/spec, recently  added features need to have an unreleased version at the moment.

File: go.mod [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
-	github.com/kubernetes-csi/csi-lib-utils v0.13.0
+	github.com/kubernetes-csi/csi-lib-utils v0.14.0
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.27.8
 	github.com/operator-framework/operator-sdk v1.29.0
 	github.com/robfig/cron/v3 v3.0.1
 	github.com/stretchr/testify v1.8.4
 	go.uber.org/zap v1.24.0
-	google.golang.org/grpc v1.55.0
+	google.golang.org/grpc v1.51.0
```
Comment (nixpanic) [inline go.mod:19]: another downgrade?

File: go.mod [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
 	google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.3.0
 	google.golang.org/protobuf v1.30.0
-	k8s.io/api v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
+	k8s.io/api v0.27.1
+	k8s.io/apimachinery v0.27.1
+	k8s.io/client-go v0.27.1
```
Comment (nixpanic) [inline go.mod:24]: not sure you really want to update the k8s dependencies....

File: apis/go.mod [hunk 4]
```diff
@@ -0,0 +1,72 @@
+module github.com/csi-addons/kubernetes-csi-addons/apis
+
+go 1.20
+
+require (
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.27.8
+	k8s.io/api v0.27.3
```
Comment (nixpanic) [inline apis/go.mod:8]: maybe use the version that is in the main `go.mod`?

File: go.mod (hunk 1 above)
Comment (subhamkrai) [reply go.mod:10]: go mod tidy is bringing all these changes.

**Review by nixpanic — APPROVED** [review]

**Comment by subhamkrai** [comment]
closing on favour of #382

**Pull request closed without merging**